tls/s2n_prf.c

```diff
@@ -452,25 +452,42 @@ int s2n_prf_calculate_master_secret(struct s2n_connection *conn, struct s2n_blob
 
     POSIX_ENSURE_EQ(s2n_conn_get_current_message_type(conn), CLIENT_KEY);
 
+    if(!conn->ems_negotiated) {
+        POSIX_GUARD(s2n_tls_prf_master_secret(conn, premaster_secret));
+        return S2N_SUCCESS;
+    }
+
     /* TODO: https://github.com/aws/s2n-tls/issues/2990 */
-    if (conn->ems_negotiated && s2n_in_unit_test()) {
-        /* Only the client writes the Client Key Exchange message */
-        if (conn->mode == S2N_CLIENT) {
-            POSIX_GUARD(s2n_handshake_finish_header(&conn->handshake.io));
-        }
-        struct s2n_stuffer client_key_message = conn->handshake.io;
-        POSIX_GUARD(s2n_stuffer_reread(&client_key_message));
-        uint8_t client_key_message_size = s2n_stuffer_data_available(&client_key_message);
-        struct s2n_blob client_key_blob = { 0 };
-        POSIX_GUARD(s2n_blob_init(&client_key_blob, client_key_message.blob.data, client_key_message_size));
-
-        uint8_t data[S2N_MAX_DIGEST_LEN] = { 0 };
-        struct s2n_blob digest = { 0 };
-        POSIX_GUARD(s2n_blob_init(&digest, data, sizeof(data)));
-        POSIX_GUARD_RESULT(s2n_prf_get_digest_for_ems(conn, &client_key_blob, &digest));
-        POSIX_GUARD_RESULT(s2n_tls_prf_extended_master_secret(conn, premaster_secret, &digest));
+    if (!S2N_IN_TEST) {
+        return S2N_SUCCESS;
+    }
```

Comment:
Wait, is a no-op right here? Shouldn't it behave as if the flag isn't enabled?

Reply:
My bad, I see what you meant now.